docgen/src/guide/Refining_hits.md

```diff
@@ -0,0 +1,552 @@
+
+export default Hit;
+```
+
+In this component we took the name, description and owner of a certain package, and render these as hits. To enable this as `hitComponent`, we will change the line in `App.js` that renders the `Hit` component:
+
+```jsx
+import React, { Component } from 'react';
+import {
+  Configure,
+  InstantSearch,
+  SearchBox,
+  Hits,
+} from 'react-instantsearch/dom';
+import Hit from './Hit'
+
+class App extends Component {
+  render() {
+    return (
+      <InstantSearch
+        appId="OFCNCOG2CU"
+        apiKey="f54e21fa3a2a0160595bb058179bfb1e"
+        indexName="npm-search"
+      >
+        <SearchBox />
+        <CurrentRefinements />
+        <Hits hitComponent={({hit}) => <Hit hit={hit} />}/>
+        <Configure
+          attributesToRetrieve={[
+            'name',
+            'description',
+            'keywords',
+            'owner.name',
+          ]}
+        />
+      </InstantSearch>
+    );
+  }
+}
+
+export default App;
+```
+
+We manually render the `hitComponent` as a function, because we want to add extra functions to the `Hit` component later. Now we will also set up two `RefinementList`s for `keywords` and `owner.name`. We do this in a container so that the difference is clear, but you can change the rendering of this however you prefer:
+
+```jsx
+import {
+  Configure,
+  InstantSearch,
+  SearchBox,
+  Hits,
+  RefinementList
+} from 'react-instantsearch/dom';
+import Hit from './Hit';
+
+class App extends Component {
+  render() {
+    return (
+      <InstantSearch
+        appId="OFCNCOG2CU"
+        apiKey="f54e21fa3a2a0160595bb058179bfb1e"
+        indexName="npm-search"
+      >
+        <SearchBox />
+        <div style={{ display: 'flex', border: '1px solid black' }}>
+          <RefinementList attributeName="keywords" withSearchBox />
+          <RefinementList attributeName="owner.name" withSearchBox />
+        </div>
+        <Hits hitComponent={({hit}) => <Hit hit={hit} />}/>
+        <Configure
+          attributesToRetrieve={[
+            'name',
+            'description',
+            'keywords',
+            'owner.name',
+          ]}
+        />
+      </InstantSearch>
+    );
+  }
+}
+
+export default App;
+```
+
+If we look at what we have now, then we see that we can select any keyword, or any owner as expected, with buttons that don't do anything yet.
+
+<iframe src="https://codesandbox.io/embed/0gJW06BAX" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+## Adding virtual RefinementLists
+
+A "virtual" RefinementList is a RefinementList, just like the ones that are using the regular widgets, but with the exception that they won't render anything. We will make a virtual RefinementList here that will take `defaultRefinement` into account when rendering. We will make a `VirtualRefinementList` component, which uses a `connectRefinementList`: 
```

Comment:
I believe we should introduce this paragraph by explaining why we need virtual widgets in the first place